data-processing-lib/python/src/data_processing/runtime/pure_python/transform_invoker.py

```diff
@@ -0,0 +1,127 @@
+# (C) Copyright IBM Corp. 2024.
+# Licensed under the Apache License, Version 2.0 (the “License”);
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#  http://www.apache.org/licenses/LICENSE-2.0
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an “AS IS” BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+################################################################################
+
+import sys
+from typing import Any
+from data_processing.utils import ParamsUtils, get_logger
+from data_processing.runtime.pure_python import PythonTransformLauncher
+from data_processing.utils import PipInstaller, TransformRuntime, TransformsConfiguration
+from data_processing.runtime.transform_launcher import AbstractTransformLauncher
+
+project = "https://github.com/IBM/data-prep-kit.git"
+logger = get_logger(__name__)
+
+
+def _import_class(name):
+    """
+    Import class by name
+    :param name: name
+    :return:
+    """
+    components = name.split('.')
+    mod = __import__(components[0])
+    for comp in components[1:]:
+        mod = getattr(mod, comp)
+    return mod
+
+
+def invoke_transform(name: str, t_class: str, launcher: AbstractTransformLauncher,
+                     input_folder: str, output_folder: str, s3_config: dict[str, Any], params: dict[str, Any]) -> bool:
+    """
+    Invoke transform
+    :param name: transform name
+    :param t_class: transform configuration class
+    :param launcher: transform launcher
+    :param input_folder: input folder (local or S3)
+    :param output_folder: output folder (local or S3)
+    :param s3_config: S3 configuration - None local data
+    :param params: transform parameters
+    :return:
+    """
+    # Create data access params
+    if s3_config is None:
```

Comment:
I think, it makes sense to add at least a warning when both S3 and local data are provided. Otherwise, somebody will have hard time to debug, why he provided local parameters, but the system uses S3.

Reply:
Input_folder, output folder are defined in both cases. They have different formats, of course, but validation is a bit painful